commit/plugin_functions.go

```diff
@@ -585,6 +585,50 @@ func validateObservedGasPrices(gasPrices []cciptypes.GasPriceChain) error {
 	return nil
 }
 
+// validateMerkleRootsState merkle roots seq nums validation by comparing with on-chain state.
+func validateMerkleRootsState(
+	ctx context.Context,
+	lggr logger.Logger,
+	report cciptypes.CommitPluginReport,
+	reader cciptypes.CCIPReader,
+) (bool, error) {
```

Comment:
Will this func gracefully fail (i.e return an error) if the node doesn't support the dest?

Reply:
if the node does not support the dest this is not called at all

Reply:
caller provides a reader